overlord/configstate/configcore/journal.go

```diff
@@ -100,10 +100,21 @@ func handleJournalConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error
 	}
 
 	if opts == nil {
-		sysd := systemd.New(dirs.GlobalRootDir, systemd.SystemMode, nil)
-		if err := sysd.Kill("systemd-journald", "USR1", ""); err != nil {
+		ver, err := systemd.Version()
+		if err != nil {
 			return err
 		}
+
+		// old systemd-journal (e.g. on core16) closes the pipes on SIGUSR1,
+		// causing SIGPIPE and restart of snapd and other services.
+		// upstream bug: https://bugs.freedesktop.org/show_bug.cgi?id=84923,
+		// therefore only tell journald to reload if it's new enough.
+		if ver >= 236 {
+			sysd := systemd.New(dirs.GlobalRootDir, systemd.SystemMode, nil)
+			if err := sysd.Kill("systemd-journald", "USR1", ""); err != nil {
+				return err
+			}
+		}
```

Comment:
Could we issue a warning instructing that journald persistence change requires a system rebstart?

Reply:
will there be anyone to observe the warning?

Reply:
I think we could, this would only make sense on `snap set` and not when we apply early config when state is not available. I'd like to explore/do this in a followup.

Reply:
This is done in #8608